### ISSUE TITLE
WRT Deprecation and the GameSDK Discord

### ISSUE THREAD

**Issue by qreeves** [discordapp/discord-rpc#303]
@msciotti Sorry if this is off topic, but I don't know where else to raise this issue. I didn't want to hijack the deprecation thread, but I am seriously concerned about the way the rules are being enforced on the GameSDK Discord server.

I had a rather unpleasant interaction with the moderators there today when all we were trying to do was give genuine support to someone who wanted to integrate rich presence in something which GameSDK does not support (below). While I understand that it is a GameSDK topical server, I don't think this attitude is warranted given the situation.

I'd like to suggest that during this overlap period that the rules either be relaxed, or a separate channel be created on the GameSDK server so that people can engage in legitimate help for those who are being left behind by the deprecation of the RPC method. I surely hope that the Discord team understands that it will take a while for toolsets to catch up with the changes, and we can't help people in the method we are being asked to until they do.

```
[15:54] KolkiesDev - Njits: Will there come a node version of the gamesdk?
[15:55] KolkiesDev - Njits: Because I want to add rpc to a electron app
[16:34] theGordHoard: @KolkiesDev - Njits it's not an official client, and technically not the game sdk, but there is https://github.com/discordjs/RPC
[16:35] KolkiesDev - Njits: rpc is deprecated
[16:36] KolkiesDev - Njits: https://i.imgur.com/VgohERL.png
[16:37] Nommiin: It still works though...
[16:37] Quin: Yeah, it uses the same basic calls and there's no plan currently to stop supporting it.
[16:37] theGordHoard: If you look at what I linked, you'll notice that it's not the same thing @KolkiesDev - Njits. This is a community made rpc client library for node.
[16:37] Quin: GameSDK has no Linux counterpart after all.
[16:38] KolkiesDev - Njits: I know what it the discordjsrpc is
[16:38] Quin: I think he just meant the API that RPC uses
[16:39] theGordHoard: That library supports the basic calls that RPC uses
[16:39] KolkiesDev - Njits: But that depends on the rpc from discord which is deprecated
[16:39] KolkiesDev - Njits: if I'm right
[16:40] Quin: When they say deprecated, they just mean they're developing GameSDK rather than continuing with the shared library called discord-rpc
[16:40] Quin: There's a whole thread about it on the GitHub issues for discord/discord-rpc
[16:40] KolkiesDev - Njits: deprecated also means it will be removed one time
[16:41] Quin: If you read the thread, they go on to say that GameSDK uses the same API calls basically
[16:41] Quin: At least for Rich Presence
[16:42] Nommiin: I really doubt it'll be removed/not supported anyway, there are tons of products that support it
[16:42] Nommiin: Removing it and forcing said products to use GameSDK is just nonsensical
[16:42] Pyxel: One way or the other this server is for the SDK, not the RPC, it may not be removed but it's no longer supported one way or the other..
[16:42] Quin: discord-rpc was just cobbled together by one guy, whereas gamesdk is a more coordinated and planned system
[16:43] KolkiesDev - Njits: So I was asking if there will come a node gamesdk one day
[16:44] Quin: So all talk and devs helping each other is banned if it involved RPC?
[16:44] Pyxel: Considering this server is for the SDK, and the RPC is no longer supported, I'd reccomend you don't go on about it here, I've had to tell someone once before to drop it.
[16:45] Pyxel: @KolkiesDev - Njits iirc not unless the devs themselves go about it due to section 2.2 of the legal documentation on the sdk.
[16:48] Quin: It's not like the community can instantly create a whole new set of tools because Discord decides to deprecate something without much warning. You need to understand that takes time and there will be an overlap period where people are going to have to use what is currently available to them.
[16:49] Quin: It's not like we're hating on GameSDK and saying RPC 4 Life
[16:49] crystalcat: Nevertheless, this server is for discussing and getting help on the Game SDK, not the RPC.
[16:50] Quin: OK, Linux version when?
[16:51] Quin: I don't want to be "that guy", but I find this attitude quite unfair.
[16:52] crystalcat: I mean. If you wanna discuss RPC, there's presumably other places. This server is the server for the GameSDK and the GameSDK only, so I'm afraid talk about it here is off topic.
[16:52] crystalcat: It's in the server rules.
[16:53] Quin: Do you guys represent Discord in an official capacity?
[16:54] Quin: I'm curious who wrote these rules and why giving genuine help is frowned upon, hardly seems in the spirit of friendly conduct.
[16:56] Pyxel: Whether we do or not does not matter. You have been told multiple times this server is not for the topic of the RPC, as stated in the rules. This is your last warning.
[16:57] Quin: I'm not talking about RPC, I'm asking a genuine question as a developer who was planning on using GameSDK. Do you represent Discord officially? My support for Discord is now contingent on the answer.
[16:59] Pyxel: I'm only a moderator under discord's rule, I am not an employee if that answers your question. If under that context, no I do not. I'm here to enforce the rules, and make sure the server stays in order.
[16:59] Quin: Thank you, have a nice day then.
```

**Comment by msciotti** [comment]
Hi there qreeves,

This question is a bit nuanced. We spun up the GameSDK help server because there was nowhere else for people to go to get help integrating the SDK. In doing so, we want to keep it _really_ on topic because we all know how things can get out of hand quickly once the first #general or #off-topic channel is created.

So, the question is whether RPC is on topic. I think my TLDR is that the original question that was asked was fine, and then it got off topic.

---

It's been almost 6 months since we deprecated this repo. The reason for deprecation was pretty straightforward: we had two SDKs that did (functionally) the same thing, and one of them we wanted to actively develop. In doing so, we _might_ have to break underlying RPC functionality (but, as you can see, we haven't) and we needed the flexibility to do so. So, we deprecated `discord-rpc` in favor of the GameSDK. Deprecation does not mean outright deletion, and in fact depending on your project, it may not necessarily mean "don't use this!". It means that we're not supporting this anymore, so it _could_ break, so use it if you're risk tolerant.

In the six months following that deprecation, a number of us have been here, on Reddit, in our support queues, etc. answering questions. The three that remain are:

1. Linux support
2. Open source/legal questions
3. IL2CPP support

And I think we've been very straightforward and honest about those things, and that you're free to use discord-rpc if the GameSDK doesn't suit your needs. These questions have (mostly) stopped being asked, as they've been answered numerous times, so I don't think anyone is scrambling overnight to replace discord-rpc.

---

I do not believe that the GameSDK help server warrants its own discord-rpc help section. We don't get support tickets about it, and no new issues pop up on this repo. Once in a blue moon, a question like the chat log you posted comes up, and I think it's answered appropriately. Creating a help section in the server would really be the opposite of deprecation. If you don't feel like the user in your chat log was appropriately helped, and had nowhere else to turn, we can have that conversation.  I, personally, don't think that's the case.

---

Lastly, I think for anyone reading this, if there are issues with the help server, we'd very much appreciate you reaching out to someone with the Admin role/staff badge and ask us the question rather than posting chat logs in a GitHub issue. Those server moderators are not on GitHub, nor are many of the Discord employees on that server. Frankly, I think I'm the only one in that member list that will see this issue. Furthermore, idunno, I don't think it's cool to put people publicly on blast with chat logs when the problem can be solved in other ways, especially when that person is not the one escalating.

---

TLDR: I think the question the user originally asked was fine, I think the answer they were given was sufficient, I think the moderator comment of "The question has been answered, let's keep it on topic moving forward" was correct, and I don't believe an official support outlet for discord-rpc is quite warranted, as it is deprecated. I'll talk internally with the team to make sure that questions like the user asked are handled appropriately in the server, as I believe this one was.

**Comment by qreeves** [comment]
To be fair, I tried to find a way to email the Discord team rather than put this on a public forum. The only person I am aware of myself is you, which is why I put this here. It was related to Discord RPC being deprecated and I don't believe people should be discouraged from helping others when there is no GameSDK alternative. The ensuing conversation was related to the deprecation of RPC and and it means to use RPC until GameSDK is supported fully on all platforms, and I don't believe it was off-topic.

That being said, you've resolved the matter to my satisfaction, which is all I wanted. You need to realise you create these chat channels not just to serve the purpose of employees helping your users, but so your users can help your users, and being overly oppressive in that regard is undesirable.

Due to the use of the chat log, I am happy for you to delete this issue, it was not my intention to "blast" it in a public manner, but it was unclear how to bring up this issue with an employee and that would probably be useful information to add to `#rules` or `#welcome`.
